### PR TITLE
refactor: result pattern in watcher.ts

### PR DESCRIPTION
**Pull request by LukasDeco**

### CODE REVIEW

File: packages/indexer/src/transaction/watcher.ts [hunk 1]
```diff
@@ -91,154 +132,156 @@ class TransactionWatcher {
...
+      // TODO: maybe i need to validate below succeeded. I can't use returning because this isn't an upsert so it could
+      //       be a no-op in the happy path
+      await db.con
+        .insert(schema.transactionWatcherTransactions)
+        .values({
+          txSig: signature,
+          slot,
+          watcherAcct: acct,
+        })
+        .onConflictDoNothing();
```
Comment (R-K-H) [inline packages/indexer/src/transaction/watcher.ts:216]: Be careful with this and postgres, it maintains an AI so if there is an ID with AI it'll continue to increase regardless.

**Review by R-K-H — APPROVED** [review]
LGTM!

File: packages/indexer/src/transaction/watcher.ts (hunk 1 above)
Comment (decartesol) [reply packages/indexer/src/transaction/watcher.ts:216]: There's no AI columns in the entire schema

**Comment by decartesol** [comment]
Closing this since #23 contains the changes here + the feedback I gave over DM.

**Pull request closed without merging**